CHG: The neuro-conda-2025b.yml file has been updated.

reillystephanie opened the pull request:
Updates include:
- restructuring of the yml file, so the conda install packages are possible for both x86 and ppc64le architectures.
- packages not available in ppc64le architectures have been moved to pip install where possible to use one yml file for different architectures.
- additional packages have been removed including qt6-main
- the pip installation of opencv-python has been replaced by a conda installation of opencv

Commit:
CHG: The neuro-conda-2025b.yml file has been updated.
Updates include:
- restructuring of the yml file, so the conda install packages are possible for both x86 and ppc64le architectures.
- packages not available in ppc64le architectures have been moved to pip install where possible to use one yml file for different architectures.
- additional packages have been removed including qt6-main
- the pip installation of opencv-python has been replaced by a conda installation of opencv

Commit:
CHG: The environment file neuro-conda-2025b.yml file contains the setup that is exected to compile on non ppc64le architectures

Commit:
NEW: A conda environment file for the neuro-conda-2025b version on ppc64le architecture machines

pantaray commented on envs/neuro-conda-2025b.yml:
```diff
@@ -2,37 +2,36 @@ name: neuro-conda-2025b
-  - mne==1.10.*
-  - mne-rsa=0.10.*
+  - mne==1.10.* 
+  - mne-rsa=0.10
```
I think it should be `mne-rsa=0.10.*`?

reillystephanie replied:
I had it originally with the * but the environment was causing issues. Not sure if that was what was causing it but checking the one that is globally installed it is also 0.10.

pantaray replied:
I think the asterisk *should* work unless conda changed something in their environment file syntax (totally possible)

reillystephanie replied:
Done

pantaray commented on envs/neuro-conda-2025b.yml:
```diff
@@ -2,37 +2,36 @@ name: neuro-conda-2025b
   - mne-bids-pipeline=1.9.*
   - mne-ari=0.1.*
-  - mne-microstates=0.3.*
+  - mne-microstates=0.3
```
same here: `0.3.*`

reillystephanie replied:
Same for this one. I can update them though as it shouldn't really impact the build. It may have been a different package totally causing the build error when building from yml

reillystephanie replied:
Done

pantaray commented on envs/neuro-conda-2025b.yml:
```diff
@@ -45,62 +44,66 @@ dependencies:
 
   - pip:
       # (1) Electrophysiology (LFP + M/EEG)
+      - eelbrain==0.41.*; platform_machine != "ppc64le"
```
I think the platform identifier can go, since we're using a dedicated env file for ppc64le

reillystephanie replied:
I am now working on setting up the ppc CI for testing it. Wasn't sure if they would be needed or if I can just specify the file, so kept them in for the moment. I will remove them, before merging the request.

pantaray replied:
Sounds good! You can point to CI to a specific env file, so these shouldn't be necessary any more

reillystephanie replied:
All ppc64le platform identifiers have been removed from the x86 environment file

pantaray commented on envs/neuro-conda-2025b.yml:
```diff
@@ -45,62 +44,66 @@ dependencies:
-      - tensorflow==2.20.*; platform_system == "Linux"
-      - tensorflow_datasets; platform_system == "Linux"
+      - braindecode==1.2.*; platform_system == "Linux" and platform_machine != "ppc64le"
+      - scikit-learn==1.8.*; platform_system == "Linux" and platform_machine != "ppc64le"
```
Same for all those here

reillystephanie replied:
Same as above

pantaray commented on envs/neuro-conda-2025b-ppc.yml:
```diff
@@ -0,0 +1,109 @@
+
+      # (1) Electrophysiology (LFP + M/EEG)
+      - mne==1.10.*
+      - mne-rsa==0.10.*; platform_machine != "ppc64le"
```
I think the platform identifier isn't necessary, since we use a dedicated env file for ppc64le

reillystephanie replied:
same as above

pantaray reviewed (changes requested):
Thanks for putting in the work to test this!

Commit:
CHG: Updated the general neuro-conda enviroment file, to remove the platform_system != ppc64le from select packages. Also included a * following the versions for mne-rsa and mne-microstates

Commit:
CHG: Removed the lines that previously included platform_machine != "ppc64le" as the packages were not available on ppc64le through conda or pip at the time of building the environment version

Commit:
CHG: Updated the workflows/test-fresh-installation.yml in order to also make use of the power VM runner for testing the ppc environments.

Commit:
CHG: Attempt to fix the error by removing the second jobs: reference in the .yml file for the runner

Commit:
CHG: Included a missing run: | at line 84, and to keep with the same format, the different scripts are included in a matrix.include list.

Commit:
CHG: Attempt to see if the files for installing actually get pulled for the power runner

Commit:
CHG: Included an alternative way to try and get the script to install, as it exists in the directory as expected

Commit:
CHG: Updated the neuro-conda-2025b-ppc.yml file to include additional versions for packages that are potentially causing issues due to new releases

Commit:
NEW: The symlink to the latest version of the ppc neuro-conda environment

Commit:
CHG: Updated to allow for the ppc CI pipeline. Changes include adding the neuro-conda-ppc-latest.yml file to the installation script

Commit:
CHG: Checking to see if the .bashrc is correct, and to see if the sourcing of conda directly is required for it to have the activate option.

Commit:
CHG: Attempting to remove the sourcing of the bashrc in teh activate environment to see if it fixes the conda error

Commit:
CHG: As both environments, regardless of architecture are called neuro-conda-2025b when they are build I have removed the env.ncEnv_power variable

Commit:
CHG: included a check for which architecture the test is running on and as a result which environment it should be testing.

Commit:
CHG: Including opencv into the list of packages that use a different name when being imported as the x86 version is called opencv-python

Commit:
CHG: editing the pkgMap dictionary to match the environment files

Commit:
CHG: Removed the ephyviewer package from the ppc64le conda environment. This is due to an error during import (also seen during testing on the ppc partitions of CoBIC)

Commit:
CHG: Multiple changes have been made to three files.
Changes to test-fresh-installation.yml file include removing the setting of python-version, and the matrix variables for the Test installation on ppc
Changes to the install.sh file in libexec include removing the if statement which includes settings depending on OS. Instead there is an if statement that sets the variable name at the start depending on the OS. Additional change include a file name for the default.yml conda environment depending on the OS.
Changes to the test_packages.py file in the tests includes removing the duplicate pkgMap dictionary as the ppc error was not caused by these packages.

Commit:
CHG: Edit to install.sh: Moved the definition of OS and mArch above the naming of the additional variables. Additional comment added later for choosing right installer

pantaray reviewed (approved):
This looks great! Thanks so much for putting in all the work!
